docs/Deploying/Debugging.md

```diff
@@ -0,0 +1,156 @@
+
+Sometimes Fleet operates without any errors but accessing a specific part of the web UI are slow. As a starting point it
+would be good to get a screenshot of the Network tab in the Developer Tools of your browser. The main data that needs to 
+be visible are: Name, Status, and Time (in Chrome's terms). 
+[Here's how to accomplish this using Google Chrome](https://developer.chrome.com/docs/devtools/network/).
+
+Besides from this, it might be good to continue with `MYSQL` and `REDIS`.
+
+Depending on the API, there will likely be followup questions about amount of data, but this would be a good point to 
+check in with Engineering.
+
+### MYSQL
+
+Most of the data needed to understand an issue in MySQL should've been gathered already by the basic data specified at 
+the beginning of this document. However, there is a chance that Fleet is running with a database user that is not 
+capable of querying the information needed. So here are the queries that would output a good first step in information
+gathering:
+
+```sql
+show engine innodb status;
+show processlist;
+```
+
+If read replicas are configured, another piece of important data is whether there has been any replication lag 
+registered.
+
+With all this gathered, it's a good time to reach out to the Engineering team.
+
+### REDIS
+
+In most cases, the data gathered at the beginning of this document should be enough to understand what might be 
+happening with Redis. However, if more details are needed, running the 
+[monitor command](https://redis.io/commands/monitor/) should shed more light in the issue.
+
+**WARNING**: if Redis is suffering from performance issues, running monitor will only increase the problem.
+
+### OSQUERY
+
+Just like with the Fleet server, the best way to understand issues on the client side is to look at logs.
+
+If you are running vanilla osquery in the host, please restart the host with `--tls_dump` and `--verbose`. This will 
+allow us to see more details as to what's happening in the communication with Fleet (or lack there of).
+
+If you are running Orbit, you should add `--debug` to the command line options. This will get debug logs for Orbit and 
+also for osquery automatically.
+
+If you are running Fleet Desktop there's no change needed, you should see the log file in the following directories 
+depending on the platform:
+
+- Linux: `$XDG_STATE_HOME/Fleet` or `$HOME/.local/state/Fleet`
+- macOS: `$HOME/Library/Logs/Fleet`
+- Windows: `%LocalAppData%/Fleet`
+
+The log file name is `fleet-desktop.log`.
+
+If the issue is related to osquery extensions, the following data would be needed:
+
+- osquery version
+- OS it's running on
+- What does the extension do?
+- How is the extension queried/deployed?
+- What language is the extension implemented in?
+
+With this data, it's time to reach out to Engineering.
+
+### `INFRA`
+
+At this level, what you want to look into are Load Balancer logs, errors, and configurations. For instance, does the LB 
```

Comment:
Would it make sense to have a link to https://health.aws.amazon.com/health/status (for AWS if you are logged in this will also redirect to a specific account health portal as well) and also a general "check your infra statuspage" type thing to cover other clouds?  Sometimes people don't even think about their cloud provider possibly having issues, and this lets them at least cross that off the list.